### ISSUE TITLE
Broadcast IP not bound to in OS X

### ISSUE THREAD

**Issue by bf1967** [JoelBender/bacpypes#68]
The broadcast address is not bound to in OS X.

In bvllservice.UDPMultiplexer.__init__ there is a check to bind to the broadcast address for non-windows machines. Part of that check is "'win' not in sys.platform". OS X reports back with a platform of 'darwin' and thus fails the check and doesn't end up bound to the broadcast address.

Here is the code in question... 

        # create and bind the broadcast address for non-Windows
        if specialBroadcast and (not noBroadcast) and 'win' not in sys.platform:
            self.broadcast = _MultiplexClient(self)
            self.broadcastPort = UDPDirector(self.addrBroadcastTuple, reuse=True)
            bind(self.direct, self.broadcastPort)
        else:
            self.broadcast = None

**Comment by ChristianTremblay** [comment]
Not sure about how you say it...
The check is there to "exclude" Windows machine from binding to the broadcast address.
So, if running on Linux or OSx, the function will try to bind the broadcast address.

That said, on windows, I was getting an error when trying to bind to the broadcast address. Do you get an error on OSx ?

**Comment by bf1967** [comment]
I added an exclusion for OS X in the affected module and submitted a pull request #69

**Comment by JoelBender** [comment]
I think what we really want is the same kind of check like I did in `task.py`, so how about changing the check to be more pro-active?  Like this:

```
        if specialBroadcast and (not noBroadcast) and sys.platform.startswith(('linux', 'darwin')):
            self.broadcast = _MultiplexClient(self)
            self.broadcastPort = UDPDirector(self.addrBroadcastTuple, reuse=True)
            bind(self.direct, self.broadcastPort)
```

**Comment by bf1967** [comment]
Is it only the unix-derivatives this check applies to?  There are 8 possible platforms that could be reported.

https://docs.python.org/2/library/sys.html

I'm not sure if BACpypes gets used in any if the others.  Otherwise I'd agree with the change you've proposed.

**Comment by JoelBender** [comment]
Should there be a check for "supported platforms" in the `__init__.py` file and if it's not one that we, the BACpypes community, recognize then we punt with a warning?  I don't know how the other platforms get broadcast packets to the appropriate socket.

**Comment by bf1967** [comment]
That would seem to be the friendliest thing to do.  Otherwise someone starts using the package assuming it works on their system of choice.   It becomes very clear which are supported/tested.

**Comment by JoelBender** [comment]
Thank you for the reference, I just created an `issue-68` branch.  Please look it over and see if it's acceptable.  Note that the code for Python 3.4 diverges from Python 2.5 and 2.7 slightly because the values for `sys.platform` changed in Python 3.3.

**Comment by bf1967** [comment]
Looks good to me.   I closed my pull request by the way.

**Comment by JoelBender** [comment]
This has been merged and released in 0.13.7
